### ISSUE TITLE
comparison with other toolkits

### ISSUE THREAD

**Issue by danpovey** [wantee/connLM#1]
@wantee, how does this compare with, say, CUED-RNNLM?  Also, what university are you at?
@hainan-xv, you might be interested to have a look at this.

**Comment by wantee** [comment]
Actually, the benchmark showed here is done like two year or earlier ago. I didn't hear about CUED-RNNLM that time. connLM would be faster than the original rnnlm by Mikolov. And I remember it should be faster and better than the faster-rnnlm, which is in the Kaldi repository in that time, but I can't  make sure of this.
I have a little experience with CUED-RNNLM recently, but did not compare them seriously. I assume that the performances(including both speed and ppl) would be close. 
For speed, CUED-RNNLM makes use of GPU and matrix multiplication, so it can't do any factorization for the softmax layer.  But connLM could do class-based factorization and ASGD with many threads(like 12 or 24). However, with large hidden size, the GPU may be more efficient, I did experiments on librispeech corpus for at most 300 hidden size, because i didn't observe significant improvement with larger hidden size but it became slower a lot. 
For ppl, I suspect that the spliced sentence bunch training used by CUED-RNNLM could introduce some performance loss. But the softmax factorization and ASGD used by connLM should also have the same problem. So I think with same size of parameters, the difference in PPL should not be large.
And also, I think CUED-RNNLM doesn't include the MaxEnt model, which may be very useful for faster convergence.
The main reason I didn't do the benchmarks is that I am not very like the current implementation in master branch. i currently working on the graph branch, which is supposed to be more flexible. Users can change the topology of network by a topo config file without changing code, with more fine control for the hyperparamters, and I plan to support more other optimization algorithms.
I am not in a university now. I worked for a company in China and my job is for speech recognition. This open source project is just a hobby for me working in my spare time(so it lasts for so long time and not finish yet).

**Comment by danpovey** [comment]
ok thanks for the info.
